lib/rust/ensogl/component/slider/src/lib.rs

```diff
@@ -326,59 +641,117 @@ impl Slider {
 
         frp::extend! { network
             background_color <- all2(&input.set_background_color, &input.set_slider_disabled);
             background_color_anim.target <+ background_color.map(desaturate_color);
             eval background_color_anim.value((color) model.set_background_color(color));
-            track_color <- all2(&input.set_slider_track_color, &input.set_slider_disabled);
-            track_color_anim.target <+ track_color.map(desaturate_color);
-            eval track_color_anim.value((color) model.set_track_color(color));
+            indicator_color <- all2(&input.set_value_indicator_color, &input.set_slider_disabled);
+            indicator_color_anim.target <+ indicator_color.map(desaturate_color);
+            eval indicator_color_anim.value((color) model.set_indicator_color(color));
             value_text_color <- all2(&input.set_value_text_color, &input.set_slider_disabled);
             value_text_color_anim.target <+ value_text_color.map(desaturate_color);
             eval value_text_color_anim.value((color) model.set_value_text_property(color));
             label_color <- all2(&input.set_label_color, &input.set_slider_disabled);
             label_color_anim.target <+ label_color.map(desaturate_color);
             eval label_color_anim.value((color) model.label.set_property_default(color));
+
+            output.disabled <+ input.set_slider_disabled;
         };
     }
 
-    /// Initialize the precision adjustment areas above/below the slider and the default precision
-    /// value.
-    fn init_precision_defaults(&self) {
+    /// Initialize the textual value editing FRP network.
+    fn init_value_editing(&self) {
+        let network = self.frp.network();
+        let input = &self.frp.input;
+        let output = &self.frp.private.output;
+        let model = &self.model;
+
+        frp::extend! { network
+            start_editing <- input.start_value_editing.constant(true).gate_not(&output.disabled);
+            start_editing <- start_editing.gate_not(&input.set_value_text_hidden);
+            output.editing <+ start_editing;
+            output.editing <+ input.finish_value_editing.constant(false);
+            output.editing <+ input.cancel_value_editing.constant(false);
+            value_on_edit <- output.value.sample(&start_editing);
+            prec_on_edit <- output.precision.sample(&start_editing);
+            max_places_on_edit <-
+                input.set_max_disp_decimal_places.sample(&start_editing);
+            value_text_on_edit <- all3(&value_on_edit, &prec_on_edit, &max_places_on_edit);
+            value_text_on_edit <- value_text_on_edit.map(|t| value_text_truncate(t).to_im_string());
+            model.value_text_edit.set_content <+ value_text_on_edit;
+            eval_ start_editing(model.set_edit_mode(true));
+            eval_ input.finish_value_editing(model.set_edit_mode(false));
+            eval_ input.cancel_value_editing(model.set_edit_mode(false));
```

Comment:
instead of so many evals, a little bit better way of doing it is in frp. You should use as few evals as possible. So instead of this you can do:

```
edit_mode_disabled <- any(&input.finish_value_editing, &input.cancel_value_editing);
new_edit_mode <- bool(& edit_mode_disabled, &start_editing);
eval new_edit_mode((t) model.set_edit_mode(*t));
```

Please apply to other places if applicable.